pkgs/development/python-modules/array-record/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, absl-py
+, etils
+, importlib-resources
+, pythonOlder
+, typing-extensions
+, zipp
+}:
+
+buildPythonPackage rec {
+  pname = "array-record";
+  version = "0.4.0";
+  format = "wheel";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit version format;
+    pname = "array_record";
+    dist = "py310";
+    python = "py310";
+    hash = "sha256-VHDU6RLR/z3/tNxJiDdAruz1cva6cHu5NzMlsKrLYXg=";
+  };
+
+  propagatedBuildInputs = [
+    absl-py
+    etils
+    importlib-resources
+    typing-extensions
+    zipp
+  ];
+
+  pythonImportsCheck = [ "array_record" ];
+
+  meta = with lib; {
+    description = "ArrayRecord is a new file format derived from Riegeli, achieving a new frontier of IO efficiency";
+    homepage = "https://github.com/google/array_record";
+    license = licenses.asl20;
```

Comment:
Do we need to add ``sourceProvenance = with sourceTypes; [ binaryNativeCode ];``?

Reply:
I have no idea :/

Reply:
I am going to fix this tree wide in the builder function.